irobot_edu_sdk/create3.py

```diff
@@ -159,7 +161,29 @@ async def navigate_to(self, x: Union[int, float], y: Union[int, float], heading:
         completer = Completer()
         self._responses[(dev, cmd, inc)] = completer
         await self._backend.write_packet(Packet(dev, cmd, inc, payload))
-        timeout = self.DEFAULT_TIMEOUT + int(math.sqrt(x * x + y * y) / 10) + 4  # 4 is the timeout for a potential rotation.
+        
+        #-----Zurcher modification (original commented out)-----
+        ''' 
+        Original line was calculating the timeout based off the passed destination coordinates, not the distance traveling.
+        When traveling a longer distance to the origin(0,0), or a destination near the origin, the navigate_to() function
+        would time out before the destination was reached.  For example, if traveling to the origin (0,0) from a location 
+        further away than the robot can transit in 7 seconds, the robot will return from navigate_to() and begin processing
+        the next function, long before the robot ever reaches (0,0).
+
+        To correct this issue, the current position is polled, and delta_x & delta_y, which represent the actual distance 
+        to the destination are calculated by subtracting the current position from the destination position.  Then, these 
+        values are used in place of X and Y in the original timeout calculation.
+        '''
+
+        self.get_position()
+        delta_x=(x-self.pose.x)
+        delta_y=(y-self.pose.y)
+        timeout = self.DEFAULT_TIMEOUT + int(math.sqrt(delta_x*delta_x + delta_y * delta_y) / 10) + 4  # 4 is the timeout for a potential rotation.
```

Comment:
Please add spaces between the first `*` as you did with the second `*` -- that is, the line should read:
```
        timeout = self.DEFAULT_TIMEOUT + int(math.sqrt(delta_x * delta_x + delta_y * delta_y) / 10) + 4  # 4 is the timeout for a potential rotation.
```